docs/concepts/rewriters.md

```diff
@@ -0,0 +1,565 @@
+
+Due to the dynamic nature of expression manipulation, rewriters are designed with interactive environments in mind. Rewriters have a `_repr_latex_` method that prints the current expression, meaning the following code in a Jupyter notebook:
+```python
+sympy_rewriter("a + b")
+```
+would return a $\LaTeX$ (technically $KaTeX$) expression $a + b$. This, combined with method chaining, means the effect of different methods can be seen immediately.
+
+
+## Concepts
+In designing the rewriter framework we implemented a number of different utility classes. A typical user should not need to interact with these objects, but we describe them here for completeness.
+
+#### Instructions
+An `Instruction` is an action that marks a change to an expression. The following `Instructions` are implemented:
+
+- `Initial`
+
+    The initial form of the rewriter instance.
+
+- `Simplify`
+
+    A backend-specific 'simplify' command.
+
+- `Expand`
+
+    Expand all brackets in the expression.
+
+- `Assumption`
+
+    Add an assumption onto a symbol.
+
+- `Substitution`
+
+    Substitute a symbol or expression for another.
+
+- `ReapplyAllAssumptions`
+
+    Reapply all previously applied assumptions.
+
+The primary purpose of `Instructions` is to track the history of an expression, and the user may only ever interact with these objects via the `history()` method. Of these classes, only `Assumption` and `Substitution` require special logic; the others have no methods or attributes implemented. We explore `Assumption` and `Substitution` in more detail below.
+
+!!! note "Why not just an `Enum`?"  
+    Originally the `Instructions` were implemented as an `Enum`! However, because `Assumptions` and `Substitutions` required special logic it was challenging to enforce strict typing across both an `Enum` and other dataclasses. Creating an empty class `Instruction` that everything else subclassed from is a cleaner implementation.
+
+
+#### Assumptions
+Assumptions about symbols can be input to the expression, and (in the case of SymPy) the backend symbolic engine attempts to simplify the expression with this new knowledge. An assumption requires three arguments:
+
+- `symbol_name: str`
+
+    The symbol to add the assumption to.
+
+- `comparator: Comparator | str`
+
+    The comparison to apply, one of ">", "<", ">=", "<=". 
+
+- `value: int | float`
+
+    The reference value to compare the symbol to.
+
+Alternatively an assumption can be parsed directly from a string:
+```python
+sympy_rewriter("max(0, X)").assume("X > 0")
+>>> X
+```
+
+Given an input assumption to a `sympy_rewriter`, the symbol is updated with the relevant SymPy [predicates](https://docs.sympy.org/latest/guides/assumptions.html#predicates). For some symbol `X` the predicates we support are:
+
+- positive: `X > a > 0` or `X > 0`,
```

Comment:
I don't get what's the difference between `X > a > 0` and `X > 0` in this context.